presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/TransformCorrelatedScalarAggregationToJoin.java

```diff
@@ -111,7 +112,7 @@ public Result apply(LateralJoinNode lateralJoinNode, Captures captures, Context
     {
         return searchFrom(rootNode, lookup)
                 .where(AggregationNode.class::isInstance)
-                .recurseOnlyWhen(isInstanceOfAny(ProjectNode.class, EnforceSingleRowNode.class))
+                .recurseOnlyWhen(isInstanceOfAny(ProjectNode.class, EnforceSingleRowNode.class, LimitNode.class))
```

Comment:
This looks unrelated to the coercion issue.  Can you put it in a separate commit (can be in the same pr) and also explain the change.

Reply:
when the subquery ends with a LIMIT clause, the transformation needs to go one further step to look for aggNode because limit could be part of the correlated subquery above the agg. sure will place in a separate commit